subkey/src/main.rs

```diff
@@ -209,6 +213,129 @@ where
 
 			print_extrinsic(extrinsic);
 		}
+		("hard-sign", Some(matches)) => {
+			let mut message = vec![];
+			stdin().lock().read_to_end(&mut message).expect("Error reading from stdin");
+			if matches.is_present("hex") {
+				message = hex::decode(&message).expect("Invalid hex in message");
+				println!("message {:#?}", message);
+			}
+			let sig = dev.hd_sign(&message);
+			match sig {
+    			Ok(v) => {
+       				println!("signature: {:?}", v);
+   				 }
+   				 Err(e) => {
+       				 println!("sign error: {:?}", e);
+    			}
+			}	 
+		}
+		("hard-format", Some(_matches)) => {
+			match  dev.hd_format() {
+    			Ok(_v) => {
+       				println!("format ok");
+   				 }
+   				 Err(e) => {
+       				 println!("format error: {:?}", e);
+    			}
+			}	 
+		}
+		("hard-gen", Some(_matches)) => {
+			 match dev.hd_generate() {
+    			Ok(v) => {
+       				println!("seed: {}",HDDevice::to_hex(&dev,&v));
+   				 }
+   				 Err(e) => {
+       				 println!("generate error: {:?}", e);
+    			}
+			}	 
+		}
+		("hard-getpub", Some(_matches)) => {
+			match dev.hd_getpub() {
+    			Ok(v) => {
+       				println!("address: 0x{}",v);
+   				 }
+   				 Err(e) => {
+       				 println!("generate error: {:?}", e);
+    			}
+			}
+			let pk:[u8;32] = dev.hd_getpub().unwrap().into();
+			println!("public: 0x{}",HDDevice::to_hex(&dev,&pk));
+		}
+		("hard-import", Some(matches)) => {
+			let mut seed = vec![];
+			stdin().lock().read_to_end(&mut seed).expect("Error reading from stdin");
+			if matches.is_present("hex") {
+				seed = hex::decode(&seed).expect("Invalid hex in message");
+				println!("seed {:#?}", seed);
+			}
+			match dev.hd_import(&seed) {
+    			Ok(_v) => {
+       				println!("import success!");
+   				 }
+   				 Err(e) => {
+       				 println!("import error: {:?}", e);
+    			}
+			}
+		}
+		("hard-transfer", Some(matches)) => {
+			let to = matches.value_of("to")
+				.expect("parameter is required; thus it can't be None; qed");
+			let to = sr25519::Public::from_string(to).ok().or_else(||
+				sr25519::Pair::from_string(to, password).ok().map(|p| p.public())
+			).expect("Invalid 'to' URI; expecting either a secret URI or a public URI.");
```

Comment:
```suggestion
let to = sr25519::Public::from_string(to)
	.ok()
	.or_else(|| sr25519::Pair::from_string(to, password).ok().map(|p| p.public()))
	.expect("Invalid 'to' URI; expecting either a secret URI or a public URI.");
```

- Don't apply suggestion as it would probably further mess the indentation. 
- Could be applied to other()
    .chained()
    .functions()